packages/protocol/contracts/common/linkedlists/SortedLinkedList.sol

```diff
@@ -71,8 +71,18 @@ library SortedLinkedList {
     bytes32 lesserKey,
     bytes32 greaterKey
   ) internal {
-    remove(list, key);
-    insert(list, key, value, lesserKey, greaterKey);
+    require(contains(list, key), "key not in list");
+    require(key != bytes32(0) && key != lesserKey && key != greaterKey, "invalid key");
+
+    bytes32 oldLesserKey = list.list.elements[key].previousKey;
```

Comment:
contains, oldLesserKey and oldGreaterKey are 3 reads of same store - it would be a bit more gas efficient to load the element into local variable and run contains check and take lesser and greater key

Reply:
Not sure what you mean exactly. Each of these slots is only read once as far as I can tell.

Reply:
Oh, do you mean reading in the Element struct, and manually performing the `contains` check? Could do that, but the gas savings there would be minimal, since we still need 3 SLOADs, so would only be saving on the hashing operations to find the mapping element.

Reply:
Currently it seems to do 2 SLOADS for every variable (load struct & load variable)

loading of struct for second and third variable should be already warm so it is rather gas efficient on the other hand we are calling this 110 times in one transaction so it could add up

Reply:
I'm pretty sure loading a struct field takes only 1 SLOAD - the compiler knows exactly where to look for it (after its found the mapping element's location by hashing its storage slot).